docs/contracts/v4/first-hook/03-hook-deployment.mdx

```diff
@@ -0,0 +1,122 @@
+---
+id: hook-deployment
+title: Hook Deployment
+sidebar_position: 3
+---
+
+# Deployment
+Deploying Uniswap v4 Hooks involves several steps:
+
+1. **Deploying the PoolManager Contract**: This contract is typically pre-deployed on many test environments. However,
+   you have the option to deploy it locally on your machine if required.
+
+2. **Deploying the Hook Contract**: The hook contract needs to be deployed at a predetermined address. You can use
+   `CREATE2` for deterministic deployment. A Deterministic Deployment Proxy, usually found
+   at `0x4e59b44847b379578588920cA78FbF26c0B4956C`, is employed for this purpose and is already available in most
+   environments.
+
+3. **Deploying Test Tokens**: These tokens are essential for creating the pool. They need to be deployed before
+   initializing the pool.
+
+4. **Initializing the Pool with the Hook Contract Address**: This is achieved by invoking
+   the `initialize(PoolKey memory key, uint160 sqrtPriceX96, bytes calldata hookData)` function on the PoolManager contract.
+
+5. **Adding Liquidity or Modifying Position**: If you wish to add liquidity to the pool or alter its position, a
+   utility contract that implements the `ILockCallback` interface is necessary. You may consider deploying a utility
+   contract like `PoolModifyPositionTest` for these operations.
```

Comment:
Wonder if we should focus more on hook deployment, and less emphasis on v4 deployments (poolmanager / periphery routers)